src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -0,0 +1,216 @@
+import UploadFileIcon from '@mui/icons-material/UploadFile';
+import FilePresentIcon from '@mui/icons-material/FilePresent';
+import * as React from 'react';
+import { type DropzoneOptions, useDropzone } from 'react-dropzone';
+import {
+	Box,
+	Link,
+	Typography,
+	CircularProgress,
+	Avatar,
+	InputLabel,
+	List,
+	ListItem,
+	ListItemText,
+} from '@mui/material';
+import type { WidgetProps } from '@rjsf/utils';
+
+export interface OnFileReadSuccessParams {
+	dataUrl: string;
+	uploadedFile: File;
+}
+
+type FileWidgetProps = WidgetProps & {
+	onChange: (params?: OnFileReadSuccessParams) => void;
+	onError?: (errors: string[]) => void;
+	value?: OnFileReadSuccessParams | string;
+};
+
+export const FileWidget = ({
+	onChange,
+	onError,
+	value,
+	...props
+}: FileWidgetProps) => {
+	const [isLoading, setIsLoading] = React.useState<boolean>(false);
+	const [loadingPercentage, setLoadingPercentage] = React.useState<number>(0);
```

Comment:
nit: We can do this with just the `loadingPercentage` state if we make it `number | undefined` (so we set it to `undefined` when it is no longer loading and when it starts loading we set it to `0`).

I raise this because these two states are directly related to each other and it would be nice to only have to maintain one state instead